controller.go

```diff
@@ -462,6 +462,7 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 	if err != nil {
 		return nil, err
 	}
+	network.generic[netlabel.IpamType] = network.ipamType
```

Comment:
I don't think the network driver should be made aware of the ipam driver in use, in line with the current model where
the ip address management and network plumbing functions are being kept separate, independent.

I think what the network driver should do is to make sure it gets the configurations it needs. 
For example, in this case the bridge driver should fail if no IPv4 Pool is passed.